### PR TITLE
[Metrics Complexity] Add workarounds for lizard's bugs

### PR DESCRIPTION
**Pull request by gnakagaw**
> Explain a summary, purpose, or background for this change.

We're facing some analysis errors due to lizard's bugs. This PR add a workaround for that.

The bugs are...
* Lizard outputs illegal XML when a function name contains [Unicode control characters](https://en.wikipedia.org/wiki/Unicode_control_characters) (e.g. BEL, BS, DEL, etc..)
* Lizard stops unexpectedly saying `RuntimeError: maximum recursion depth exceeded error` with a complex file analysis.

I forked [the original repo](https://github.com/terryyin/lizard/) in Sider Organization and [add some workarounds commits](https://github.com/terryyin/lizard/compare/master...sider:add-some-improvement). `Metrics Complexity` runner uses the workaround version in this PR.

NOTE: I'm preparing to send patches to lizard project. After the patches merged, we can remove these workarounds.

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ Misc:
 - Introduce common option `linter.<id>.dependencies` [#2211](https://github.com/sider/runners/pull/2211)
 - **Metrics File Info** Report the number of commits, additions, deletions used to measure code churn [#2184](https://github.com/sider/runners/pull/2184)
 - **ESLint** Allow strings for ext and global options [#2236](https://github.com/sider/runners/pull/2236)
+- **Metrics Complexity** add workarounds for lizard's bug [#2249](https://github.com/sider/runners/pull/2249)
```
Comment (ybiquitous) [inline CHANGELOG.md:45]: [trivial] Seems a typo:

```suggestion
- **Metrics Complexity** Add workarounds for lizard's bug [#2249](https://github.com/sider/runners/pull/2249)
```

File: test/smokes/metrics_complexity/expectations.rb [hunk 2]
```diff
@@ -193,3 +193,39 @@
...
+      message: :_,
+      object: :_,
+      links: [],
+      git_blame_info: nil
+    }
+  ],
+  analyzer: { name: "Metrics Complexity", version: default_version}
+)
+
+#TODO: This is a test case to ensure avoiding lizard's bug. Remove after the bug fixed.
```
Comment (ybiquitous) [inline test/smokes/metrics_complexity/expectations.rb:215]: Also here:

```suggestion
```

**Review by ybiquitous — APPROVED** [review]
LGTM 👍 

Besides, could you create an issue to address this TODO, please?

**Comment by ybiquitous** [comment]
I've trimmed a period `.` from the PR title (the title is not a sentence).

**Comment by ybiquitous** [comment]
@yykamei ping?

**Comment by gnakagaw** [comment]
@ybiquitous We're discussing here about lizard code.
https://github.com/sider/lizard/commit/9525f66b7ade3e03891fd63981a96d93d1c9b94f#commitcomment-48971358

**Comment by ybiquitous** [comment]
Oh, sorry. I didn't know it 😅 

Do you want to include this PR in the next release of Runners? If not, I will not be hurried.

**Comment by gnakagaw** [comment]
@ybiquitous
> Do you want to include this PR in the next release of Runners?

No. I need some work to modify lizard code and runner code. No need to wait this PR.

**Comment by ybiquitous** [comment]
OK 👍

**Comment by gnakagaw** [comment]
I added `DO NOT MERGE` label to this issue. I'm doing a refactoring the workaround python code with yykamei. I'll remove the tag when we finished the task.
https://github.com/sider/lizard/pull/1

**Review by yykamei — APPROVED** [review]
LGTM

**Comment by gnakagaw** [comment]
@ybiquitous 
I've finished editing Python code. I also changed runner's code a little. Can you check that?

This is the major modification. We add `--max-recursion-depth` parameter to lizard.
https://github.com/sider/runners/pull/2249/commits/8330f43bd39d99a6e941ddad57aa2a3f1d4effe5

**Review by ybiquitous — APPROVED** [review]
LGTM 👍 

Please rebase `master` before the merge.

**Pull request merged**